app/forms/candidate_interface/completion_form.rb

```diff
@@ -0,0 +1,32 @@
+module CandidateInterface
+  class CompletionForm
+    include ActiveModel::Model
+
+    def self.build_from_application(attr_hash)
+      attr = attr_hash.keys.first
+
+      create_attr_accessor_from(attr)
+      create_validation_from(attr)
+
+      new(attr_hash)
+    end
+
+    def self.create_attr_accessor_from(attr)
+      instance_eval do
+        attr_accessor attr
+      end
```

Comment:
I'm not sure this `instance_eval` is doing what you expect(?). It adds the attribute to the class rather than the instance (and the relevant instance hasn't been created yet).

```
[1] pry(main)> f = CandidateInterface::CompletionForm.build_from_application(foo: 'foo')
=> #<CandidateInterface::CompletionForm:0x00007fa0994a4bb0 @foo="foo">
[2] pry(main)> f.foo
=> "foo"
[3] pry(main)> b = CandidateInterface::CompletionForm.build_from_application(bar: 'bar')
=> #<CandidateInterface::CompletionForm:0x00007fa099567d18 @bar="bar">
[4] pry(main)> b.foo
=> nil
[5] pry(main)> b.respond_to?(:foo)
=> true
[6] pry(main)> f.respond_to?(:bar)
=> true
[7] pry(main)>
```

I guess this works anyway because the `CandidateInterface::CompletionForm` will be given the attributes it needs before it needs them and it doesn't matter that it accumulates them all over time because they won't interfere.

But I'm not sure about the validation, as soon as you add a second attribute the form will never be valid...

```
[8] pry(main)> f.valid?
=> false
from (pry):8:in `<main>'
[9] pry(main)> f.errors.full_messages
=> ["Bar can't be blank"]
```